dragonfly/implementations/daq_run_interface.py

```diff
@@ -332,4 +343,613 @@ def save_trace(self, trace, comment):
+        result = self.provider.cmd(self.psyllid_interface, 'quit_psyllid', payload = self.payload_channel)
+        result = self.provider.cmd(self.daq_target, 'unblock_channel', payload=self.payload_channel)
+
+#    def _set_condition(self, number):
+#        if number in self._set_condition_list:
+#            logger.debug('deactivate psyllid daq')
+#            result = self.provider.cmd(self.psyllid_interface, 'activate', payload = self.payload_channel)
+#        elif number == 0:
+#            logger.debug('getting out of safe mode')
+#        else:
+#            logger.debug('condition {} is unknown: ignoring!'.format(number))
+
+
+# frequency settings and gettings
+
+    def _get_roach_central_freqs(self):
+        result = self.provider.cmd(self.daq_target, 'get_all_central_frequencies')
+        logger.info('ROACH central freqs {}'.format(result))
+        return result
+
+
+    def _get_psyllid_central_freqs(self):
+        result = self.provider.cmd(self.psyllid_interface, 'get_all_central_frequencies')
+        logger.info('Psyllid central freqs {}'.format(result))
+        return result
+
+
+    @property
+    def central_frequency(self):
+        return self.freq_dict[self.channel_id]
+
+
+    @central_frequency.setter
+    def central_frequency(self, cf):
+        payload={'cf':cf, 'channel':self.channel_id}
+        result = self.provider.cmd(self.daq_target, 'set_central_frequency', payload=payload)
+        # The roach frequency can differ from the requested frequency. Psyllid should have the same frequency settings as the ROACH
+        self.freq_dict[self.channel_id]=result['values'][0]
+        payload={'channel':self.channel_id, 'cf':self.freq_dict[self.channel_id], 'channel':self.channel_id}
+        result = self.provider.cmd(self.psyllid_interface, 'set_central_frequency', payload=payload)
+        if result['values'][0]!=True:
+            logger.warning('Could not set central frequency in Psyllid')
+            self.freq_dict[self.channel_id]=None
+            raise core.exceptions.DriplineGenericDAQError('Could not set central frequency in Psyllid stream')
+
+
+    def make_trigger_mask(self, snr = 11, filename = '/home/roach/trigger_masks/psyllid_trigger_mask'):
+        # Set threshold snr
+        payload = {'channel':self.channel_id, 'snr':snr}
+        result = self.provider.cmd(self.psyllid_interface, 'set_fmt_snr_threshold', payload = payload)
+        time.sleep(1)
+        # Acquire and save new mask
+        timestr = time.strftime("%Y%m%d-%H%M%S")
+        filename = filename+'_channel_'+self.channel_id+'_'+timestr+'.json'
+        payload = {'channel':self.channel_id, 'filename':filename}
+        result = self.provider.cmd(self.psyllid_interface, 'make_trigger_mask', payload=payload)
+        if result['values'][0]!=True:
+            return False
+        else: return True
+
+
+
+
+__all__.append('ROACHMultiChAcquisitionInterface')
+class ROACHMultiChAcquisitionInterface(DAQProvider, core.Spime):
```

Comment:
I would like to make a new feature for the multi channel classes and remove them from this branch. I havent worked on them since march and they are not ready for usage.
I would suggest to leave the single channel classes where they are right now and when the multi channel classes will be added to develop one day we can move them all into a new source file.